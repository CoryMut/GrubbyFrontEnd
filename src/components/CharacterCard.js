import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";

const useStyles = makeStyles(() => ({
    lightCard: {
        cursor: "pointer",
        backgroundColor: "lightgray",
        margin: "0.5vw 0.5vw 0 0",
    },
    darkCard: {
        cursor: "pointer",
        backgroundColor: "green",
        margin: "0.5vw 0.5vw 0 0",
    },
}));

const CharacterCard = ({ name, handleCharacter, characters }) => {
    const classes = useStyles();
    let className = classes.lightCard;

    // if (characters.includes(name)) {
    //     className = classes.darkCard;
    // }

    return (
        <Button className={className} onClick={() => handleCharacter(name)}>
            {name}
        </Button>
    );
};

export default CharacterCard;
