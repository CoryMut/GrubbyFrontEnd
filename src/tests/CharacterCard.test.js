import React from "react";
import { shallow } from "enzyme";
import CharacterCard from "../components/CharacterCard";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

test("should test NavBar component", () => {
    const wrapper = shallow(<CharacterCard />);
    expect(wrapper).toMatchSnapshot();
});

test("it renders without crashing", () => {
    render(
        <MemoryRouter>
            <CharacterCard />
        </MemoryRouter>
    );
});
