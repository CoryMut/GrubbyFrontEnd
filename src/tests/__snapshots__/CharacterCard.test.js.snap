// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test NavBar component 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-lightCard-1"
  onClick={[Function]}
/>
`;
